pilot/pkg/model/config.go

```diff
@@ -683,30 +684,24 @@ func (store *istioConfigStore) ExternalServices() []Config {
 // change event and pass them to the config generator. Model calls to List are
 // extremely expensive - and for larger number of services it doesn't make sense
 // to just convert again and again, for each listener times endpoints.
-func (store *istioConfigStore) VirtualServices(gateways []string) []Config {
+func (store *istioConfigStore) VirtualServices(gateways map[string]bool) []Config {
 	configs, err := store.List(VirtualService.Type, NamespaceAll)
 	if err != nil {
 		return nil
 	}
 
-	// Trim the list to virtual services bound to the gateways specified
-
-	gatewayRequested := make(map[string]bool)
-	for _, gateway := range gateways {
-		gatewayRequested[gateway] = true
-	}
-
 	out := make([]Config, 0)
 	for _, config := range configs {
 		rule := config.Spec.(*networking.VirtualService)
 		if len(rule.Gateways) == 0 {
 			// This rule applies only to IstioMeshGateway
-			if gatewayRequested[IstioMeshGateway] {
+			if gateways[IstioMeshGateway] {
 				out = append(out, config)
 			}
 		} else {
-			for _, ruleGateway := range rule.Gateways {
-				if gatewayRequested[ruleGateway] {
+			log.Infof("service %v has gateways %v, comparing against %v", config.Name, rule.Gateways, gateways)
```

Comment:
I assume you are going to get rid of these printfs?

Reply:
Ah, missed that one. Tried to move them all to `Debugf` at least.